pkg/executor/statefulset.go

```diff
@@ -63,21 +56,27 @@ func UpdateStatefulSetReplicas(
 	cr *kdv1.KubeDirectorCluster,
 	replicas int32,
 	statefulSet *appsv1.StatefulSet,
+	client k8sclient.Client,
 ) error {
 
-	replicasPatch := []replicasPatchSpec{
-		{
-			Op:    "replace",
-			Path:  "/spec/replicas",
-			Value: replicas,
-		},
-	}
-	replicasPatchBytes, patchErr := json.Marshal(replicasPatch)
-	if patchErr == nil {
-		patchErr = sdk.Patch(statefulSet, types.JSONPatchType, replicasPatchBytes)
+	// TODO: can we just use the statefulset that was passed in?
```

Comment:
I think so. It either comes from observer.GetStatefulSet in initRoleInfo, or from executor.CreateStatefulSet in handleRoleCreate. Probably shouldn't re-fetch it here, to avoid consistency issues. Hopefully the new form of CreateStatefulSet is still returning a complete valid statefulset object.

And of course we really should move back to using a PATCH whenever we can move to a stable version of controller-runtime that supports that (v0.2.0 or later). We can note that in a new Issue.

Reply:
More discussion of PATCH stuff in my long comment above.